rama-tcp/src/pool/mod.rs

```diff
@@ -0,0 +1,182 @@
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+use rama_core::error::OpaqueError;
+use rand::{seq::SliceRandom, RngCore};
+
+use crate::client::TcpStreamConnector;
+
+/// Selection algorithms
+#[derive(Debug, Clone)]
+pub enum Selector {
```

Comment:
`ThreadRng` is `!Send` & `!Sync`, therefore I did not store it on the selector.

If creating a new `ThreadRng` each time is too much overhead we could store a [random iterator](https://rust-random.github.io/rand/rand/trait.Rng.html#method.random_iter) and then manually impl `Clone` for `Selector`. Something like

```rust

impl Clone for Selector {
   fn clone(&self) -> Self {
    match self {
      Self::Random(iter) -> Self::Random(iter.cloned())
      Self::RoundRobin(atomic_usize) -> Self::RoundRobin(atomic_usize.clone())
    }
   }
}

```